pkg/trace/traceutil/trace.go

```diff
@@ -104,6 +104,12 @@ func ComputeTopLevel(trace pb.Trace) {
 			SetTopLevel(span, true)
 			continue
 		}
+		_, isOtelTrace := span.Meta["otel.trace_id"]
```

Comment:
Hmmm, this is a bit odd and leaks OTEL details into the core loop of the agent in "top level" detection. Is there any reason you can't just mark the trace as "Client Computed Top Level" and leave this function un-modified? (Note that the caller of this function in `process` skips this call if the trace is marked as ClientComputedTopLevel)

Reply:
We can do this but we still want to apply the root span logic of ComputeTopLevel for OTel spans. If we move the root span logic into the OTel pipeline it would result in some duplicate code, but I'm not sure which option is preferable

Reply:
Just updated my PR to remove changes to `ComputeTopLevel` and move the logic into a function in otlp.go `computeTopLevelAndMeasured`.